File: code/design-patterns/structural/decorator/python/decorator_usage.py
from decorator import (
    EmailNotifier,
    SMSNotifier,
    FacebookNotifier,
    SlackNotifier,
    SimpleNotifier,
)

# Wrap the basic notifier with multiple decorators
notifier = SimpleNotifier()
notifier = EmailNotifier(SMSNotifier(FacebookNotifier(SlackNotifier(notifier))))

notifier.send()

File: code/design-patterns/structural/decorator/go/decorator_usage.go
package main

// Example usage
func main() {
	// Wrap the basic notifier with multiple decorators
	var notifier Notifier = &SimpleNotifier{}
	notifier = &EmailNotifier{notifier}
	notifier = &SMSNotifier{notifier}
	notifier = &FacebookNotifier{notifier}
	notifier = &SlackNotifier{notifier}
	notifier.Send()
}
